lib/Dialect/LWE/Conversions/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -431,12 +435,12 @@ struct ConvertRMul : public OpConversionPattern<RMulOp> {
     if (xT.getNumElements() != 2 || yT.getNumElements() != 2) {
       op.emitError() << "`lwe.rmul` expects ciphertext as two polynomials, got "
                      << xT.getNumElements() << " and " << yT.getNumElements();
-      return failure();
+      return rewriter.notifyMatchFailure(op, "ciphertext not two polynomials");
     }
 
     if (xT.getElementType() != yT.getElementType()) {
       op->emitOpError() << "`lwe.rmul` expects operands of the same type";
-      return failure();
+      return rewriter.notifyMatchFailure(op, "operands not of same type");
```

Comment:
this one and the above also call `emitOpError` or `emitError` and I think it's probably more worth kicking that out and keeping the more verbose notifyMatchFailure instead